### PR TITLE
Various psalm fixes in files_sharing

### PR DESCRIPTION
**Pull request by CarlSchwan**
See individual commits fore more details

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -581,7 +581,7 @@ public function createShare(
 			}
 
 			// TODO: It might make sense to have a dedicated setting to allow/deny converting link shares into federated ones
-			if (($permissions & Constants::PERMISSION_READ) && $this->shareManager->outgoingServer2ServerSharesAllowed()) {
+			if ($this->shareManager->outgoingServer2ServerSharesAllowed()) {
```
Comment (come-nc) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:584]: Why is this removed?

**Review by come-nc — APPROVED** [review]

File: apps/files_sharing/lib/Controller/ShareAPIController.php (hunk 1 above)
Comment (CarlSchwan) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:584]: because permissions is always at least set to read just before, so this was always true

**Review by skjnldsv — APPROVED** [review]
